### PR TITLE
Proper support for multiple queues inside one MQ

### PR DESCRIPTION
**Pull request by Almad**
# Merge ALL the things!!11!!11

[Cloud Dredd runs in queues (fix upstream ivy bug)](https://trello.com/c/YlMbBQRp)
Fixed queue name propagation.

Blocking & pending:

* [x] actually listen to multiple queues
* [x] multiple queues in memory backend

Refactors pending (might not happen in first iteration):

* [ ] queue as in ironmq -> mq in the whole domain
* [ ] protect from multiple setupQueue calls
* [ ] sendOptions vs. receivingOptions (instead of one options dict)
* [ ] queue clear array tangle
* [ ] fix taskExecuted backchannel calls & emits to include queue name

### CODE REVIEW

**Comment by Almad** [comment]
//cc @abtris @kuba-kubula

File: src/queues/ironmq.coffee [hunk 1]
```diff
@@ -58,12 +65,25 @@ class IronMQQueue
     if options.immediatePush
       @consumeTasks()
 
-  clear: (cb) ->
-    @queue.del_queue (err, body) ->
+  clear: (queues, cb) ->
+    console.error 'queues is', queues
```
Comment (abtris) [inline src/queues/ironmq.coffee]: leftover?

File: src/queues/ironmq.coffee [hunk 2]
```diff
@@ -58,12 +65,25 @@ class IronMQQueue
...
+    if typeof queues is 'function'
+      cb     = queues
+      queues = (name for name, q of @queues)
+
+    async.forEach queues, (name, done) =>
+      console.error 'clearing', name
```
Comment (abtris) [inline src/queues/ironmq.coffee]: leftover?

File: README.md [hunk 3]
```diff
@@ -41,13 +41,12 @@ ivy.registerTask(factorial, finished, {
...
 
   ivy.setupQueue({
-      queue: 'testpackage',
-
       type: 'ironmq',
       auth: {
           token:      process.env.IRONMQ_TOKEN      || 'dummy',
           project_id: process.env.IRONMQ_PROJECT_ID || 'testpackage'
       }
+      //, queue: 'testpackage' // optional, inferred from task
```
Comment (abtris) [inline README.md]: Better remove?

File: docker-compose.yml [hunk 4]
```diff
@@ -0,0 +1,26 @@
+data:
+  image: busybox
+  volumes:
+     - /tmp/ivydata/ironmq:/ironmq/db
+     - /tmp/data/redis:/data/
+ironmq:
+  image: iron/mq
```
Comment (abtris) [inline docker-compose.yml]: WOW! Good to know!

**Comment by abtris** [comment]
Nothing blocking for me now.

File: README.md (hunk 3 above)
Comment (Almad) [inline README.md]: Nope, this is `README` as in documentation and I plan to support it

File: docker-compose.yml (hunk 4 above)
Comment (Almad) [inline docker-compose.yml]: Yup, reason why we are using them — see [docs](http://dev.iron.io/mq/3/on-premise/installation/)

Comment (w-vi) [inline docker-compose.yml]: I like it, glad to hear we are going to stick to it.

**Pull request merged**